R/rjournal_article.R

```diff
@@ -1,30 +1,127 @@
 #' R Journal format.
 #'
 #' Format for creating R Journal articles. Adapted from
-#' \url{https://journal.r-project.org/submissions.html}.
-#' @param ...,citation_package Arguments to \code{rmarkdown::pdf_document}.
+#' <https://journal.r-project.org/submissions.html>.
+#'
+#' @details
+#'
+#' This file is only a basic article template. For full details of _The R
+#' Journal_ style and information on how to prepare your article for submission,
+#' see the [Instructions for Authors](https://journal.r-project.org/share/author-guide.pdf)
+#'
+#' ## About this format and the R Journal requirements
+#'
+#' `rticles::rjournal_article` will help you build the correct files requirements:
+#'
+#' - A R file will be generated automatically using `knitr::purl` - see
+#' https://bookdown.org/yihui/rmarkdown-cookbook/purl.html for more information.
+#' - A tex file will be generated from this Rmd file and correctly included in
+#' `RJwapper.tex` as expected to build `RJwrapper.pdf`.
+#' - All figure files will be kept in the default rmarkdown `*_files` folder. This
+#' happens because `keep_tex = TRUE` by default in `rticles::rjournal_article`
+#' - Only the bib filename is to modifed. An example bib file is included in the
+#' template (`RJreferences.bib`) and you will have to name your bib file as the
+#' tex, R, and pdf files.
+#'
+#' # About YAML header fields
+#'
+#' This section documents some of the YAML fields that can be used with this
+#' formats.
+#'
+#'
+#' ## The `author` field in the YAML header
+#'
+#' | FIELD  | TYPE | DESCRIPTION |
+#' | ------ | ---- | ----------- |
+#' | `name` | *required* | name and surname of the author |
+#' | `affiliation` | *required* | name of the author's affiliation |
+#' | `address` | *required* | at least one address line for the affiliation |
+#' | `url` | *optional* | an additional url for the author or the main affiliation |
+#' | `orcid` | *optional* | the authors ORCID if available |
+#' | `email` | *required* | the author's e-mail address |
+#' | `affiliation2` | *optional* | name of the author's 2nd affiliation |
+#' | `address2` | *optional* | address lines belonging to the author's 2nd affiliation |
+#'
+#' *Please note: Only one `url`, `orcid` and `email` can be provided per author.*
+#'
+#' ## Other YAML fields
+#'
+#' | FIELD | TYPE | DESCRIPTION |
+#' | ----- | ---- | ----------- |
+#' | `bibliography` | *with default* | the BibTeX file with the reference entries |
+#'
+#' @inheritParams rmarkdown::pdf_document
+#' @param ... Arguments to `rmarkdown::pdf_document.
+#'
+#' @md
```

Comment:
I feel it may be a better idea to put the main documentation in `skeleton.Rmd`, because I'm not sure if authors would actually read the help page. What do you think?

Reply:
Personally I think that it is cleaner to have it in the documentation of the function (as it is). The documentation in the RMD-file for the R Journal is already neat and intelligible. I would not add too much more here. My understanding is that `'rticles'` supports the writing process, so less technical details might be better.

Reply:
The part **About this format and the R Journal requirements** is already in the skeleton. 
FOr the **The author field in the YAML header** section, I'll add a mention in the skeleton to consult the documentation.